src/main/java/com/backbase/ct/bbfuel/data/PaymentsDataGenerator.java

```diff
@@ -132,4 +142,19 @@ private static Bank generateCreditorBank() {
                 .withCountry(faker.address().countryCode())
                 .withCountrySubDivision(faker.address().state()));
     }
+
+    private static Bank generateCreditorBankWithBicCode() {
```

Comment:
duplicating method: either call `generateCreditorBank().withBic(getRandomFromList(bicCodes))` or just add the `.withBic(getRandomFromList(bicCodes))` to the existing `generateCreditorBank()` method. There is no harm done because all banks do have a BIC

Reply:
done